apps/passport-client/components/shared/PCDCardList.tsx

```diff
@@ -0,0 +1,279 @@
+  const sortablePCDs = useMemo(
+    () =>
+      pcds.map((pcd, i) => ({
+        pcd,
+        name: pcdCollection.getPackage(pcd.type).getDisplayOptions?.(pcd)
+          ?.displayName,
+        order: i
+      })),
+    [pcdCollection, pcds]
+  );
+  // only show sort options that are valid for all PCDs
+  const sortOptions = useMemo(
+    () =>
+      SORT_OPTIONS.filter(
+        (option) =>
+          !sortablePCDs.some((pcd) => typeof pcd[option.key] === "undefined")
+      ),
+    [sortablePCDs]
+  );
+  const [sortState, setSortState] = useState<SortState>({
+    sortBy: "order",
+    sortOrder: "asc"
+  });
+  const sortedPCDs = useMemo(
+    () =>
+      (sortState.sortBy && sortState.sortOrder
+        ? _.orderBy(sortablePCDs, [sortState.sortBy], [sortState.sortOrder])
+        : sortablePCDs
+      ).map((o) => o.pcd),
+    [sortState, sortablePCDs]
+  );
+
+  const [selectedPCDID, setSelectedPCDID] = useState("");
+  const selectedPCD = useMemo(() => {
+    // if user just added a PCD, highlight that one
+    if (sessionStorage.newAddedPCDID != null) {
+      const added = pcds.find((pcd) => pcd.id === sessionStorage.newAddedPCDID);
+      if (added) {
+        return added;
+      }
+    }
+
+    const selected = pcds.find((pcd) => pcd.id === selectedPCDID);
+    if (selected) {
+      return selected;
+    }
+
+    // default to first PCD if no selected PCD found
+    return pcds[0];
+  }, [pcds, selectedPCDID]);
+
+  const onClick = useCallback((id: string) => {
+    setSelectedPCDID(id);
+  }, []);
+
+  return (
+    <Container>
+      {sortablePCDs.length > 1 && (
+        <ToolBar
+          sortOptions={sortOptions}
+          sortState={sortState}
+          onSortStateChange={setSortState}
+        />
+      )}
+      {sortedPCDs.map((pcd) => (
+        <PCDCard
+          key={pcd.id}
+          pcd={pcd}
+          isMainIdentity={pcd.id === mainPCDId}
+          onClick={onClick}
+          expanded={pcd.id === selectedPCD?.id}
+        />
+      ))}
+    </Container>
+  );
+}
+
+function ToolBar({
+  sortOptions,
+  sortState,
+  onSortStateChange
+}: {
+  sortOptions: SortOption[];
+  sortState: SortState;
+  onSortStateChange: (sortState: SortState) => void;
+}) {
+  if (sortOptions.length === 0) return null;
+
+  return (
+    <ToolBarContainer>
+      {sortOptions.map((o) => {
+        const active = sortState.sortBy === o.key;
+        const onClick = () => {
+          if (active) {
+            onSortStateChange({
+              sortBy: o.key,
+              sortOrder: sortState.sortOrder === "asc" ? "desc" : "asc"
+            });
+          } else {
+            onSortStateChange({
+              sortBy: o.key,
+              sortOrder: "asc"
+            });
+          }
+        };
+
+        return (
+          <ToolBarItem key={o.key} onClick={onClick}>
+            {o.render()}
+            <SortIcon sortOrder={active ? sortState.sortOrder : undefined} />
+          </ToolBarItem>
+        );
```

Comment:
may be worth making these icons bigger - remember that they're mainly going to be used on a browser. another thing that could be nice is making use of the `IconButton` thing we've got going on on the modal `X` icon for example - it gives a bit more user feedback by having a its background change to a semi-transparent circle when the user clicks on it.